src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -3172,18 +3181,9 @@ bool hasImplicitConversion(TypeSymbol source, TypeSymbol destination, ref Compou
 
         private BetterResult BetterParamsCollectionType(TypeSymbol t1, TypeSymbol t2, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
         {
-            CollectionExpressionTypeKind kind1 = ConversionsBase.GetCollectionExpressionTypeKind(Compilation, t1, out TypeWithAnnotations elementType1);
-            CollectionExpressionTypeKind kind2 = ConversionsBase.GetCollectionExpressionTypeKind(Compilation, t2, out TypeWithAnnotations elementType2);
-
-            if (kind1 is CollectionExpressionTypeKind.CollectionBuilder or CollectionExpressionTypeKind.ImplementsIEnumerable)
-            {
-                _binder.TryGetCollectionIterationType(CSharpSyntaxTree.Dummy.GetRoot(), t1, out elementType1);
-            }
-
-            if (kind2 is CollectionExpressionTypeKind.CollectionBuilder or CollectionExpressionTypeKind.ImplementsIEnumerable)
-            {
-                _binder.TryGetCollectionIterationType(CSharpSyntaxTree.Dummy.GetRoot(), t2, out elementType2);
-            }
+            var syntax = CSharpSyntaxTree.Dummy.GetRoot();
+            ConversionsBase.TryGetCollectionExpressionTypeKind(_binder, syntax, t1, out CollectionExpressionTypeKind kind1, out TypeWithAnnotations elementType1);
```

Comment:
From offline discussion, maybe this change should be reverted and the `TryGetCollectionExpressionTypeKind` could be removed/inlined to the three calling locations (conversions, conversion errors and params binding).

Reply:
I've removed `TryGetCollectionExpressionTypeKind` and updated the callers. The callers now have minimal changes from the feature branch, so it may make sense to review those locations from the baseline rather than as a diff between commits. Apologies for the code churn.